### PR TITLE
chore: add contribution guidelines

### PR DESCRIPTION
**Pull request by bavulapati**
This contribution guidelines will help the new contributors to get started with the pull requests for micro tasks. We will update these guidelines, as we make progress through the project.

### CODE REVIEW

**Review by RaisinTen — DISMISSED** [review]
This LGTM but it would be nice if we could also provide some project specific resources that would help potential contributors to update themselves on the skills and technologies required for this project. I'm okay with adding that info in this PR or a subsequent PR, you choice!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Welcome to OpenAPI Componentizer contributing guide <!-- omit in toc -->
+
+Thank you for investing your time in contributing to our project!
+
+Read our [Code of Conduct](./CODE_OF_CONDUCT.md) to keep our community approachable and respectable.
+
+We will begin this guide to get new contributors to get started with pull requests and we will update this guide as we progress through the project.
+
```
[the remark this answers is not in the document]
Comment (bavulapati) [reply CONTRIBUTING.md:8]: @RaisinTen This would inform the contributors that this is a WIP guide and they will have to revisit this doc in future. WDYT?

**Review by arvindkalra08 — DISMISSED** [review]

**Comment by bavulapati** [comment]
> This LGTM but it would be nice if we could also provide some project specific resources that would help potential contributors to update themselves on the skills and technologies required for this project. I'm okay with adding that info in this PR or a subsequent PR, you choice!

The project specific details are added in a separate [PR](https://github.com/postman-open-technologies/openapi-componentizer/pull/4).

**Comment by bavulapati** [comment]
@arvindkalra08 @RaisinTen @MikeRalphson We currently have the following requirements for a PR:
1. Needs to approvals
2. When the PR is updated, the approvals get invalidated.

WDYT about these requirements?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+## New contributor guide
+
+To get an overview of the project, read the [README](README.md). Here are some resources to help you get started with open source contributions:
+
+- [Finding ways to contribute to open source on GitHub](https://docs.github.com/en/get-started/exploring-projects-on-github/finding-ways-to-contribute-to-open-source-on-github)
+- [Set up Git](https://docs.github.com/en/get-started/quickstart/set-up-git)
+- [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow)
+- [Collaborating with pull requests](https://docs.github.com/en/github/collaborating-with-pull-requests)
+
+* Each feature/API/PR should be supported with adequate tests and documentation. Documentation and tests are first class citizens of the project. This helps the project survive in the long run.*
```
Comment (RaisinTen) [inline CONTRIBUTING.md:16]: Extra `*` at the end:
```suggestion
* Each feature/API/PR should be supported with adequate tests and documentation. Documentation and tests are first class citizens of the project. This helps the project survive in the long run.
```

**Review by RaisinTen — APPROVED** [review]
LGTM

**Comment by RaisinTen** [comment]
> Needs to approvals

2 approvals SGTM

> When the PR is updated, the approvals get invalidated.

Sometimes one might approve a PR but also suggest some optional nits and addressing those probably doesn't require another approval (like I did right here on this PR - https://github.com/postman-open-technologies/openapi-componentizer/pull/3#pullrequestreview-1348316753), so I don't think this is ideal but no objections from me if anyone of you see value in this one.

**Comment by bavulapati** [comment]
I already feel annoyed about the second requirement. 
If I accept the suggestion now, I would have to request your approval again as that would make changes to the PR.

**Comment by bavulapati** [comment]
Removed the second rule. 
Reason: https://github.com/postman-open-technologies/openapi-componentizer/pull/3#issuecomment-1476080023

Let me know if you think, the removed rule adds any value.

**Review by MikeRalphson — APPROVED** [review]
LGTM

**Pull request merged**